cellmonitor/source/android/app/src/main/java/org/kaaproject/kaa/demo/cellmonitor/MainActivity.java

```diff
@@ -243,6 +266,29 @@ private void sendLog() {
                 latitude, longitude);
     }
 
+    @Override
+    public void onRequestPermissionsResult(int requestCode,
+                                           @NonNull String permissions[], @NonNull int[] grantResults) {
+        switch (requestCode) {
+            case PERMISSION_LOCATION_REQUEST_CODE: {
+                if (grantResults.length > 0
+                        && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+
+                    locationManagerWrapper = new LocationManagerWrapper(this, cellCallback);
+                    cellManager = new CellManager(this, cellCallback);
+                    checkLocationEnabled();
+                    updateContent();
```

Comment:
But this functionality is repeated with similar from onCreate. This method is more convinient